packages/compass-schema/src/components/schema-toolbar/schema-toolbar.tsx

```diff
@@ -130,14 +131,37 @@ const SchemaToolbar: React.FunctionComponent<SchemaToolbarProps> = ({
           </div>
         </div>
       )}
-      {analysisState === ANALYSIS_STATE_ERROR && (
+      {error?.errorType === 'GENERAL' && (
         <ErrorSummary
           data-testid="schema-toolbar-error-message"
-          errors={[`${ERROR_WARNING}: ${errorMessage}`]}
+          errors={[`${ERROR_WARNING}: ${error.errorMessage}`]}
+          dismissible={true}
+          onClose={onDismissError}
+        />
+      )}
+      {error?.errorType === 'TIMEOUT' && (
+        <WarningSummary
+          data-testid="schema-toolbar-timeout-message"
+          warnings={[INCREASE_MAX_TIME_MS_HINT_MESSAGE]}
+          dismissible={true}
+          onClose={onDismissError}
         />
       )}
-      {analysisState === ANALYSIS_STATE_TIMEOUT && (
-        <WarningSummary warnings={[INCREASE_MAX_TIME_MS_HINT_MESSAGE]} />
+      {error?.errorType === 'HIGH_COMPLEXITY' && (
+        <Banner
+          variant={BannerVariant.Danger}
+          data-testid="schema-toolbar-complexity-abort-message"
+          dismissible={true}
+          onClose={onDismissError}
+        >
+          The analysis was aborted because the number of fields exceeds 1000.
```

Comment:
Where did we get this 1000 from? I'm mostly wondering if 10,000 would perform well.

Could we also have a constant to share this number with the `distinctFieldsAbortThreshold` option? That way we're sure they're kept in sync if we do change it.